addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -298,12 +298,32 @@ odoo.define('website.s_website_form', function (require) {
                     }
                     switch (successMode) {
                         case 'redirect': {
-                            successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
+                            const hashtagIndex = successPage.indexOf("#");
+                            if (hashtagIndex > 0) {
+                                // Compare the redirect link (with and without
+                                // the lang url code) with the current page to
+                                // know if if we should stay on the same page
+                                // in the case of an anchor link.
+                                const target = successPage.replaceAll("/", "");
+                                if ([target, session.lang_url_code + target].some(link => link.startsWith(window.location.pathname.replaceAll("/", "") + '#'))) {
+                                    successPage = successPage.substring(hashtagIndex);
+                                }
+                            }
                             if (successPage.charAt(0) === "#") {
-                                await dom.scrollTo($(successPage)[0], {
-                                    duration: 500,
-                                    extraOffset: 0,
-                                });
+                                // Escape special characters to make the jQuery
+                                // selector to work.
+                                const $successPage = $('#' + $.escapeSelector(successPage.substring(1)));
```

Comment:
Not unlike the above, this looks like a rather complicated way of spelling out `document.getElementById` (unless the page can have multiple ids with the same value but that's illegal).

Reply:
indeed